File: BaseLibrary/command/miner/PutMineCommand.cs
using System.Collections.Generic;

namespace BaseLibrary.command.miner {
    public class PutMineCommand : AMinerCommand{

        private static readonly List<ISubCommandFactory> SUB_COMMAND_FACTORIES = new List<ISubCommandFactory>();

        public static int RegisterSubCommandFactory(ISubCommandFactory subCommandFactory) {
            int position = SUB_COMMAND_FACTORIES.Count;
            SUB_COMMAND_FACTORIES.Add(subCommandFactory);
            return position;
        }
    }
}

File: Java-Client/RobotBattlefield-Client-Java/src/BaseLibrary/communication/command/miner/PutMineCommand.java

package BaseLibrary.communication.command.miner;

import BaseLibrary.communication.protocol.AProtocol;

public class PutMineCommand extends AMinerCommand {

	private static final String NAME = "PUT_MINE";

	static {
		AProtocol.registerForDeserialize(NAME, PutMineCommand.class);
	}

	@Override
	public String getCommandName() {
		return NAME;
	}
}
